google/cloud/firestore_v1/stream_generator.py

```diff
@@ -14,21 +14,50 @@
 
 """Classes for iterating over stream results for the Google Cloud Firestore API.
 """
+from __future__ import annotations
 
-from collections import abc
+from typing import TYPE_CHECKING, Any, Generator, Optional
 
+from google.cloud.firestore_v1.query_profile import (
+    ExplainMetrics,
+    QueryExplainError,
+)
 
-class StreamGenerator(abc.Generator):
-    """Generator for the streamed results."""
+if TYPE_CHECKING:  # pragma: NO COVER
+    from google.cloud.firestore_v1.query_profile import ExplainOptions
 
-    def __init__(self, response_generator):
+
+class StreamGenerator(Generator[Any, Any, Optional[ExplainMetrics]]):
```

Comment:
1. We should use a generic for the output type, instead of Any
2. We probably don't need to return Optional[ExplainMetrics] to the end user in the API contract, we can treat that as an implementation detail. Users should access explain_metrics by calling the method

Reply:
I think you are right about both. I will update accordingly.